code/modules/antagonists/wizard/equipment/artefact.dm

```diff
@@ -299,78 +299,73 @@
 	heal_burn = 25
 	heal_oxy = 25
 
-//Warp Whistle: Provides uncontrolled long distance teleportation.
-/obj/item/warpwhistle
+///Warp whistle, spawns a tornado that teleports you
+/obj/item/warp_whistle
 	name = "warp whistle"
-	desc = "One toot on this whistle will send you to a far away land!"
+	desc = "Calls a cloud to come pick you up and drop you at a random location on the station."
 	icon = 'icons/obj/wizard.dmi'
 	icon_state = "whistle"
-	var/on_cooldown = 0 //0: usable, 1: in use, 2: on cooldown
-	var/mob/living/carbon/last_user
 
-/obj/item/warpwhistle/proc/interrupted(mob/living/carbon/user)
-	if(!user || QDELETED(src) || user.notransform)
-		on_cooldown = FALSE
-		return TRUE
-	return FALSE
+	/// Cooldown between whistle uses.
+	COOLDOWN_DECLARE(whistle_cooldown)
+	/// Person using the warp whistle
+	var/mob/living/whistler
 
-/obj/item/warpwhistle/proc/end_effect(mob/living/carbon/user)
-	user.invisibility = initial(user.invisibility)
-	user.status_flags &= ~GODMODE
-	REMOVE_TRAIT(user, TRAIT_IMMOBILIZED, WARPWHISTLE_TRAIT)
-
-
-/obj/item/warpwhistle/attack_self(mob/living/carbon/user)
-	if(!istype(user) || on_cooldown)
-		return
-	on_cooldown = TRUE
-	last_user = user
-	var/turf/T = get_turf(user)
-	playsound(T,'sound/magic/warpwhistle.ogg', 200, TRUE)
-	ADD_TRAIT(user, TRAIT_IMMOBILIZED, WARPWHISTLE_TRAIT)
```

Comment:
As you can see here, the old code was applying TRAIT_IMMOBILIZED.

this not working isn't the warp whistle's fault, it's TRAIT_IMMOBILIZED's fault

just fix that instead

Reply:
this was previous functionality

Tornado not even showing up because it can't go through walls:
https://user-images.githubusercontent.com/53777086/168824431-0420675b-fca8-45ed-bad8-279a0e214fab.mp4

Tornado working despite not even being on the same plane of existance:
https://user-images.githubusercontent.com/53777086/168824440-6263bfaa-1f29-44fd-9e60-4e4e6e032258.mp4

Reply:
How can you look at this then sleep soundly at night (pointing out the sleep too)
![image](https://user-images.githubusercontent.com/53777086/168824804-84d1a5c5-d614-4b2a-8693-fd817c655436.png)

Reply:
el `notransform`